code/datums/helper_datums/teleport.dm

```diff
@@ -198,27 +198,8 @@
 		var/z = pick(zlevels)
 		var/random_location = locate(x,y,z)
 
-		if(safety_turf(random_location))
-		// DING! You have passed the gauntlet, and are "probably" safe.
+		if(isfloorturf(random_location))
 			var/turf/simulated/floor/F = random_location
```

Comment:
You can most likely just cast it to `/turf` and then call `safety_turf`

Reply:
I don't quite understand that part

Reply:
Your new proc is defined on `/turf` level. So you can call it using a `/turf` object. `random_location` should always be a `/turf` object (however it isn't casted as one currently, just `var/random_location`).
If you change it to `var/turf/random_location` then you can use `if(random_location.is_safe())` without casting it to `/turf/simulated/floor`